### PR TITLE
Document FHIR Server Configuration Steps

### PR DESCRIPTION
**Pull request by csandersdev**
I created a new page in the User's Guide for FHIR Server Configuration and it includes everything I can think of regarding FHIR server setup. There is no mention of the VSAC IG JAR because it is no longer needed with the shift in focus to user managed value sets. A lot of the detail in Tim's Box note regarding TLS, Certificate Management, and Kubernetes was abbreviated because it is specific to internal IBM processes.

### CODE REVIEW

File: docs/user-guide/fhir-server-config.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+## Enable searching for ValueSet resources vs. those defined in an IG
+In the fhir-server-config.json for the knowledge tenant created above, set fhirServer/core/serverRegistryResourceProviderEnabled to true to enable user-defined ValueSet resources to be searchable via the REST API. This is a required setting to enable search functionality in the cohort and measure authoring tool. See [the Dev Guide](/dev-guide/value-sets?id=fhir-server-setup) for complete details. The property is also described in the [IBM FHIR Server User's Guide](https://ibm.github.io/FHIR/guides/FHIRServerUsersGuide/).
+
+## Configure the FHIR Server's SSL certificate
+IBM FHIR comes preconfigured with only an SSL endpoint enabled. A clear text HTTP endpoint can be setup, but it is *strongly recommend* not to do this for any type of production environment, especially one that processes PHI data.
```
Comment (msargentibm) [inline docs/user-guide/fhir-server-config.md:13]:
```suggestion
IBM FHIR comes preconfigured with only an SSL endpoint enabled. A clear text HTTP endpoint can be setup, but it is *strongly recommended* not to do this for any type of production environment, especially one that processes PHI data. 
```

File: docs/user-guide/fhir-server-config.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+## Configure authentication (Optional)
+IBM FHIR comes preconfigured for basic authentication using a well-known username and password. While not strictly required, it is recommended that users update their FHIR server configuration to use more secure values. Oauth is also supported. See the [IBM FHIR Server User's Guide](https://ibm.github.io/FHIR/guides/FHIRServerUsersGuide/#53-openid-connect-and-oauth-20) for details on how to configure oauth.
```
Comment (msargentibm) [inline docs/user-guide/fhir-server-config.md:20]:
```suggestion
IBM FHIR comes preconfigured for basic authentication using a well-known username and password. While not strictly required, it is recommended that users update their FHIR server configuration to use more secure values. OAuth is also supported. See the [IBM FHIR Server User's Guide](https://ibm.github.io/FHIR/guides/FHIRServerUsersGuide/#53-openid-connect-and-oauth-20) for details on how to configure OAuth. 
```

**Review by msargentibm — APPROVED** [review]
Content seemed good overall. I suggested a couple minor tweaks.

File: docs/user-guide/fhir-server-config.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+## Enable searching for ValueSet resources vs. those defined in an IG
+In the fhir-server-config.json for the knowledge tenant created above, set fhirServer/core/serverRegistryResourceProviderEnabled to true to enable user-defined ValueSet resources to be searchable via the REST API. This is a required setting to enable search functionality in the cohort and measure authoring tool. See [the Dev Guide](/dev-guide/value-sets?id=fhir-server-setup) for complete details. The property is also described in the [IBM FHIR Server User's Guide](https://ibm.github.io/FHIR/guides/FHIRServerUsersGuide/).
```
Comment (jaigoradia) [inline docs/user-guide/fhir-server-config.md:10]: the link for "the Dev Guide" is not found. Gets a 404 when clicking on it.

File: docs/user-guide/fhir-server-config.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+When running IBM FHIR on IBM Cloud it is recommended to use a certificate generated by the Kubernetes certicates API. That process is described in the [Manage TLS Certificates in a Cluster](https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/) section of the K8S user guide. Once you have a valid certificate request approved and issued in your cluster, you will need to configure  the IBM FHIR server to use that certificate and then configure the cohorting JVM to trust the server's public key. Both are typically done through Helm charts that are used to deploy the services to the IBM Cloud via a devops pipeline. Details are left up to the user, but internal teams may contact the cohorting component team for additional details on what is done within our team to meet IBM internal process guidelines.
```
Comment (jaigoradia) [inline docs/user-guide/fhir-server-config.md:17]: Spelling of certificates is incorrect.  "... generated by Kubernetes certificates API."

**Review by jaigoradia — APPROVED** [review]
Added two small comments. One spelling and the other a broken link.   Otherwise LGTM.

File: docs/user-guide/fhir-server-config.md (hunk 4 above)
Comment (tabishop) [inline docs/user-guide/fhir-server-config.md:17]: Could be useful to add a link to the FHIR server's documentation about configuring TLS  https://ibm.github.io/FHIR/guides/FHIRServerUsersGuide#52-keystores-truststores-and-the-fhir-server

**Review by tabishop — APPROVED** [review]
One minor comment, but looks good.

File: docs/user-guide/fhir-server-config.md (hunk 3 above)
Comment (csandersdev) [reply docs/user-guide/fhir-server-config.md:10]: That isn't going to work unless it is processed by docsify first. You can test by checking out the branch and running ``docify serve docs`` from the root of the project if you are interested. I just did it to make sure and it was fine.

**Pull request merged**